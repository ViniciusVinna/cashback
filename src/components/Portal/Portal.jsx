import ReactDOM from 'react-dom';
import { usePortal } from 'hooks';

const Portal = ({ id, children }) => {
  const target = usePortal(id);

  return ReactDOM.createPortal(children, target);
};

export default Portal;
