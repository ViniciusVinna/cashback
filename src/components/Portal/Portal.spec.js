import React from 'react';
import ReactDOM from 'react-dom';

import Portal from './Portal';

describe('Portal', () => {
  describe('styles', () => {
    it('should render with default styles', () => {
      ReactDOM.createPortal = node => node;
      const { container } = render(<Portal>child</Portal>);
      expect(container).toMatchSnapshot();
    });
  });
});
