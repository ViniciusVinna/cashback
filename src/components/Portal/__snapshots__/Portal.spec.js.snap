// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal styles should render with default styles 1`] = `
<div>
  child
</div>
`;
